ocaml/xapi-guard/lib/server_interface.ml

```diff
@@ -152,7 +152,26 @@ let update_key key state t =
 
 let empty = ""
 
-let serve_forever_lwt_callback_vtpm ~cache mutex vtpm _path _ req body =
+let serve_forever_lwt_callback_vtpm ~cache mutex vm_uuid _ req body =
+  let get_vtpm_ref () =
+    let* vm =
+      with_xapi ~cache @@ Xen_api_lwt_unix.VM.get_by_uuid ~uuid:vm_uuid
+    in
+    let* vTPMs = with_xapi ~cache @@ Xen_api_lwt_unix.VM.get_VTPMs ~self:vm in
+    match vTPMs with
+    | [] ->
+        D.warn
+          "%s: received a request from a VM that has no VTPM associated, \
+           ignoring request"
+          __FUNCTION__ ;
+        let msg =
+          Printf.sprintf "No VTPM associated with VM %s, nothing to do" vm_uuid
+        in
+        raise (Failure msg)
+    | self :: _ ->
+        let* uuid = with_xapi ~cache @@ Xen_api_lwt_unix.VTPM.get_uuid ~self in
```

Comment:
If the roundtrips become a problem we could add some new API calls to XAPI that do all 3 of these in one go,
e.g. an API call that takes a VM uuid and returns the contents of its first and only TPM.

Reply:
But we don't need to do that right now.